zio-kafka/src/main/scala/zio/kafka/consumer/SubscriptionStreamControl.scala

```diff
@@ -0,0 +1,32 @@
+package zio.kafka.consumer
+import zio.UIO
+
+/**
+ * Allows graceful shutdown of a stream, where no more records are being fetched but the in-flight records can continue
+ * to be processed and their offsets committed.
+ *
+ * As long as this object is in scope, the Kafka consumer remains subscribed.
+ *
+ * @tparam StreamType
+ *   Type of the stream returned from [[stream]]
+ */
+trait SubscriptionStreamControl[StreamType] {
+
+  /**
+   * The stream of partitions / records for this subscription
+   */
+  def stream: StreamType
+
+  /**
+   * Stop fetching records for the subscribed topic-partitions and end the associated streams, while allowing commits to
+   * proceed (consumer remains subscribed)
+   */
+  def stop: UIO[Unit]
+}
+
+object SubscriptionStreamControl {
+  def apply[T](stream0: T, stop0: UIO[Unit]): SubscriptionStreamControl[T] = new SubscriptionStreamControl[T] {
+    override def stream: T       = stream0
+    override def stop: UIO[Unit] = stop0
+  }
```

Comment:
`T` is java style. The scala style guide follows Haskell and recommends to use single letters starting with `A`, then `B`, etc.

However, here we can use `S` because the type is supposed to be a stream.


```suggestion
  def apply[S](stream0: S, stop0: UIO[Unit]): SubscriptionStreamControl[S] = new SubscriptionStreamControl[S] {
    override def stream: S       = stream0
    override def stop: UIO[Unit] = stop0
  }
```